src/core_plugins/kibana/public/dashboard/selectors/embeddables.js

```diff
@@ -0,0 +1,10 @@
+import {
+  getTitle,
+  getEditUrl,
+  getError,
+} from './embeddable';
+
+export const getEmbeddable = (state, panelId) => state[panelId];
+export const getEmbeddableTitle = (state, panelId) => getTitle(getEmbeddable(state, panelId));
```

Comment:
This is called `state`, but it looks like none of these things work if you pass in the entire state object directly. Am I right? If so, to me it seems like this really isn't a "selector", it's a helper.

Personally, I don't expose selectors unless they take the whole state object.

Reply:
I thought calling it state was the convention but I can call it embeddables I guess, that would make it clearer?  cc @chrisronline or @kjbekkelund - how is management calling this?

I think I need to export these mid-level selectors so the top level selectors can use them.  Maybe I am misunderstanding the convention around selectors?

Reply:
If I follow how it's being used correctly, this isn't the whole state object, but instead `state.embeddables`. So if I, externally, do `getEmbeddableTitle(state)`, this won't work. I realize I'm not suppose to import this selector like that by convention, but my point is that, at least to me, all selectors should take the *entire* state object, not just a part of it, so to me, this isn't a selector.

Reply:
This is why I kept these functions in the reducer files for Cloud: they receive _that_ part of the state so they don't need knowledge of the full state object.

Reply:
That's another way to do it too. I'm guessing that in Cloud, you aren't passing the state into the selector then, right? 

Part of why I follow the "full state" rule is because my selectors are always used in `mapStateToProps`, which gets the *entire* state object, so that's what I pass in to the selector to get what I need out of the state.

Reply:
We built them up gradually, starting here: https://github.com/elastic/cloud/blob/master/cloud-ui/public/reducers/index.js#L145-L146 (which uses https://github.com/elastic/cloud/blob/master/cloud-ui/public/reducers/allocators/index.js#L34, which knows the details about the location in the "local part of the tree"). That way the top-level helper knows where it's put in the full state tree, while we "gradually" know more specific locations. That worked well in that code base, but some might see it as "not DRY enough" or "too much boilerplate" since there's more than a single function.

Reply:
So in `mapStateToProps` you always use the functions that receive full state.

Reply:
So that code is doing kind of the same thing as this PR does, where it slices off part of the state and passes it into another selector. The idea of isolating the state in the reducer isn't bad though. It's tricky here because selectors don't live along with reducers anymore.

I've been keeping selectors separate from reducers, so nothing imports reducers (except the main store). The side effect is that selectors area pretty easy to understand, they all get the same thing, the entire state tree.